tests/common/type_list.h

```diff
@@ -33,37 +33,137 @@ struct user_struct {
   }
 };
 
+namespace user_def_types {
+// A user-defined struct with several scalar member variables, no constructor,
+//  destructor or member functions.
+struct no_cnstr {
+  float a;
+  int b;
+  char c;
+
+  void operator=(const int &v) {
+    this->a = v;
+    this->b = v;
+    this->c = v;
+  }
```

Comment:
Does this compiles without returning a reference to `*this`?
```suggestion
  auto& operator=(const int &v) {
    a = v;
    b = v;
    c = v;
    return *this;
  }
```
Use case:
```c++
no_cnstr a, b;
a = b = 2;
```